osf/management/commands/deactivate_requested_accounts.py

```diff
@@ -0,0 +1,76 @@
+import logging
+
+from website import mails
+from django.utils import timezone
+
+from framework.celery_tasks import app as celery_app
+from website.app import setup_django
+setup_django()
+from osf.models import OSFUser
+from website.settings import OSF_SUPPORT_EMAIL, OSF_CONTACT_EMAIL
+from django.core.management.base import BaseCommand
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def deactivate_requested_accounts(dry_run=True):
+    users = OSFUser.objects.filter(requested_deactivation=True, contacted_deactivation=False, date_disabled__isnull=True)
+
+    for user in users:
+        if user.has_resources:
+            mails.send_mail(
+                to_addr=OSF_SUPPORT_EMAIL,
+                mail=mails.REQUEST_DEACTIVATION,
+                user=user,
+                can_change_preferences=False,
+            )
+        else:
+            user.disable_account()
+            user.is_registered = False
+            mails.send_mail(
+                to_addr=user.username,
+                mail=mails.REQUEST_DEACTIVATION_COMPLETE,
+                user=user,
+                contact_email=OSF_CONTACT_EMAIL,
+                can_change_preferences=False,
+            )
+
+        user.contacted_deactivation = True
+        user.email_last_sent = timezone.now()
+        if not dry_run:
+            user.save()
```

Comment:
I'd wrap more of this in a dry_run - if it's a `dry_run`, you're still sending emails to users, just not saving the information, which could lead to confusion.  I'd recommend reworking so on a dry_run, so no emails are sent, no fields are changed or saved.  Perhaps a dry_run just logs a list of users with resources, and a list without so we can double check these users are correct before running it for real.